zap/src/output/mod.rs

```diff
@@ -1,40 +1,87 @@
 use std::fmt::Display;
 
-use crate::{
-	irgen::{gen_ser, Expr, Stmt, Var},
-	parser::{Ty, TyDecl},
-	util::NumTy,
-};
+use crate::{irgen::Stmt, parser::Ty, util::NumTy};
 
 pub mod client;
 pub mod server;
 
-impl Display for TyDecl {
-	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-		let name = &self.name;
-		let ty = &self.ty;
-
-		writeln!(f, "export type {name} = {ty}")?;
+pub trait Output {
+	fn push(&mut self, s: &str);
+	fn indent(&mut self);
+	fn dedent(&mut self);
```

Comment:
`dedent` feels very unclear as a function name

Reply:
It's the only thing I could think of that's two letters (please don't ask me to make it outdent :c). As a side not python also uses it?

Reply:
`rmindent`? I'm not sure how much I like that either though...
maybe `dedent` is the best?

Reply:
Let's keep it as `dedent`.